tgui/packages/tgui/interfaces/DecalPainter.tsx

```diff
@@ -52,82 +52,88 @@ export const DecalPainter = (props) => {
   const preview_color = custom_color_selected ? 'custom' : data.current_color;
 
   return (
-    <Window width={550} height={400}>
+    <Window width={525} height={400}>
       <Window.Content>
-        <Section title="Decal Color">
-          {data.color_list.map((color) => {
-            return (
-              <Button
-                key={color.color}
-                selected={color.color === data.current_color}
-                onClick={() =>
-                  act('select color', {
-                    color: color.color,
-                  })
-                }
-              >
-                <ColorBox color={filterBoxColor(color.color)} mr={0.5} />
-                {color.name}
-              </Button>
-            );
-          })}
-          {supports_custom_color && (
-            <Button
-              selected={custom_color_selected}
-              onClick={() => act('pick custom color')}
-            >
-              <ColorBox color={data.current_custom_color} mr={0.5} />
-              Custom
-            </Button>
-          )}
-        </Section>
-        <Section title="Decal Style">
-          <Flex direction="row" wrap="nowrap" align="fill" justify="fill">
-            {data.decal_list.map((decal) => {
-              const nondirectional = data.nondirectional_decals.includes(
-                decal.decal,
-              );
-
-              return nondirectional ? (
-                // Tallll button for nondirectional
-                <IconButton
-                  key={decal.decal}
-                  decal={decal.decal}
-                  dir={2}
-                  color={preview_color}
-                  label={decal.name}
-                  selected={decal.decal === data.current_decal}
-                />
-              ) : (
-                // 4 buttons for directional
-                <Flex
-                  key={decal.decal}
-                  direction="column"
-                  wrap="nowrap"
-                  align="fill"
-                  justify="fill"
+        <Stack fill vertical>
+          <Stack.Item>
+            <Section title="Decal Color">
+              {data.color_list.map((color) => {
+                return (
+                  <Button
+                    key={color.color}
+                    selected={color.color === data.current_color}
```

Comment:
everything `data.x` should be deconstructed before the return block, it's free

Reply:
Same as the other comment, I don't know what this means (I can vaguely READ UI code but have no clue about terms or how to make it work from scratch)

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring

Reply:
Before I commit, does this look right?
![image](https://github.com/user-attachments/assets/9f374231-fd8e-4618-af34-f41e126b76a0)

Reply:
ohhhh I get it, should be right now with https://github.com/tgstation/tgstation/pull/89883/commits/6ccd285311c359a05aa80c146cbbee5acb4901cf